### ISSUE TITLE
va-file-input-multiple Accessibility fixes

### ISSUE THREAD

**Issue by rsmithadhoc** [department-of-veterans-affairs/vets-design-system-documentation#2988]
## Description

Accessibility fixes for the file input - multiple component.

## Details

- **Axe** - _**Form elements must have labels**_ - See "Label and hint text not read out by screen readers" right below this.
    - `<input id="fileInputField" class="file-input" type="file" name="my-file-input-0" style="visibility: unset;">`
- **Label and hint text not read out by screen readers (Safari + VoiceOver, Chrome + JAWS, Firefox + NVDA) when focusing the element.**
    ![label-screen-reader](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/01012bed-ecf9-4795-9187-1e8d0c0d1e1c)
    - It currently only says "Choose File: No file chosen" or "No file selected".
    - I believe this has to do with the label issue mentioned above. It doesn't look like there is any ARIA attributes that create a relationship between the label, hint, and input. Not that it would work anyway, it looks like we have the label and hint text in the outer Shadow DOM, while the input is in its own Shadow DOM. We may have to duplicate the label and hint text and pass it in, add aria-labelledby/aria-describedby, and visually hide the duplicated text.
    - We need to ensure that when a file is added and another upload field gets appended, that it continues to read the label and hint when it is tabbed to.

- **Bug - After adding a file, then deleting it, it shows two "Drag files here" inputs.**
    ![multiple-drag-areas-bug](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/524e453e-ee74-42ce-8bda-366c89c8c065)

- **"Accept" variation - When dragging a non-PDF, the error is shown only with a colored border.** Color should not be the only visual way to convey meaning. There is also nothing read out when an error occurs. I believe an error message with an alert role under the hint text is appropriate here.
    <img width="790" alt="error-accept-1" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/8a784155-5520-455c-894f-b9cd5c7556fd">
    ![accept-variation-error](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/8cc20b40-e828-471e-84de-d6dcb3df01be)

- **Delete modal - focus should be trapped in the modal.** The `va-modal` does this, but I'm not sure why that isn't happening for this one.
    ![modal-focus-trap](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/fc5c494f-e9e0-4a32-83ac-ae43bc554c5a)
    - I recall seeing something saying this modal will be going away after it was requested to be added, but I can't find the issue, so I could be mistaken. So we may need to confirm that.

- ~~**Screen reader (Safari + VoiceOver, Chrome + JAWS, Firefox + NVDA) - Upload, change, or delete - no success message for any of these operations.**~~ _**Note:**_ There is a [separate issue for this](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2947)
    ![upload-change-delete-alert](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/cc9f9ed9-77e8-4f7b-8fb7-6575310897e2)

## Tasks
- [ ] Fix accessibility issues
- [ ] Re-test to ensure they work as expected

## Acceptance Criteria
- [ ] Accessibility issues have been fixed

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/2988?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @nickjg231 @powellkerry @rmessina1010 @rsmithadhoc
